lib/Transforms/MemoryBanking.cpp

```diff
@@ -509,7 +528,11 @@ void MemoryBankingPass::runOnOperation() {
       auto [it, inserted] =
           memoryToBanks.insert(std::make_pair(memrefVal, SmallVector<Value>{}));
       if (inserted)
-        it->second = createBanks(memrefVal, bankingFactor, bankingDimension);
+        it->second = createBanks(
+            memrefVal, bankingFactor,
+            bankingDimension.hasValue()
+                ? std::optional<unsigned>(bankingDimension.getValue())
```

Comment:
Why can't this just be `bankingDimension`?

Reply:
indeed it can be!